torch/testing/_internal/common_nn.py

```diff
@@ -4881,103 +4948,103 @@ def _do_test(self, test_case, module, input):
             output_ip.backward(grad)
             test_case.assertEqual(input.grad, input_ip.grad)
 
-        if isinstance(input, torch.LongTensor) and TEST_CUDA:
+        if any(isinstance(t, torch.LongTensor) for t in input_tuple) and TEST_CUDA:
```

Comment:
In this PR, we've extended NN module tests to support inputs that are tuples. However, I don't think this conditional covers some cases. At a high level, what we're trying to test here is a cartesian product of the following conditions
- .cpu(), .cuda(), and .cuda(1) work on the model and inputs
- .float() and .double() and .half() work on the model and the floating-point inputs

Let's say there exists a module that takes in (LongTensor, LongTensor, DoubleTensor) inputs (it exists -- nn.EmbeddingBag. However, nn.EmbeddingBag doesn't actually get tested in this way because of the previous lack of tupled input support for NewModuleTest). Then in the current logic we wouldn't test the following cases:
- convert (LongTensor, LongTensor, DoubleTensor) to (LongTensor, LongTensor, FloatTensor)
- convert (LongTensor, LongTensor, DoubleTensor) to (LongTensor, LongTensor, HalfTensor)
- convert (LongTensor, LongTensor, DoubleTensor) to (cuda.LongTensor, cuda.LongTensor, cuda.FloatTensor)
- convert (LongTensor, LongTensor, DoubleTensor) to (cuda.LongTensor, cuda.LongTensor, cuda.HalfTensor)

Reply:
@zou3519  You are right, this is a place I was wondering whether I should use all instead of any. nn.EmbeddingBad should be supported once input_dict is supported. But if I change this to all we probably can support it now and still keep all previous cases running correctly.
Let me change this to all, agree?

Reply:
> So we actually don't care about whetehr we can convert for eg. from (LongTensor, LongTensor, DoubleTensor) to (cuda.LongTensor, cuda.LongTensor, cuda.FloatTensor), what we care about here is that whether LongTensor can be casted to FloatTensor correctly on cpu/gpu.

I disagree, I think the point of the test is to convert all floating-point inputs but not convert any integral tensor inputs.

Consider nn.EmbeddingBag. It takes in up to [three arguments in the forward pass](https://pytorch.org/docs/stable/generated/torch.nn.EmbeddingBag.html).
```
model = nn.EmbeddingBag()
model(input: LongTensor, offsets: LongTensor, per_index_weights: Tensor)
```
If we changed `input` and/or `offsets` to FloatTensor, the model would break. This is because EmbeddingBag expects `input` and `offsets` to be a LongTensor but doesn't care what per_index_weights is. Most models that want an integral tensor require that tensor to be integral.

Reply:
@zou3519 You are partially right in my opinion. The code has two branches.
If input is LongTensor and TEST_CUDA. It actually test whether LongTensor can be convert to FloatTensor correctly by cuda. In this case, it does convert every tensor to Float.
The other branch is like what you said and I already changed the code in my previous PR, only convert non-Long Tensor.

Reply:
So for the first branch, now we only go into this branch if all input tensors are LongTensor.